src/shared/typescriptLambdaHandlerSearch.ts

```diff
@@ -35,59 +37,200 @@ export class TypescriptLambdaHandlerSearch implements LambdaHandlerSearch {
 
         const sourceFile = ts.createSourceFile(filename, fileContents, ts.ScriptTarget.Latest, true)
 
-        const handlers: LambdaHandlerCandidate[] = []
-
-        sourceFile.forEachChild(
-            childNode => {
-                const foundHandlers = TypescriptLambdaHandlerSearch.visitSourceFileChild(
-                    childNode,
-                    filename
-                )
-                handlers.push(...foundHandlers)
-            }
-        )
+        const handlers: LambdaHandlerCandidate[] = this.processSourceFile(sourceFile, filename)
 
         return handlers
     }
 
     /**
-     * @description looks for Lambda Handler candidates in given node.
+     * @description looks for Lambda Handler candidates in the given source file
      * Lambda Handler candidates are top level exported methods/functions.
      *
-     * @param node SourceFile child node to visit
+     * @param sourceFile SourceFile child node to process
      * @param filename filename of the loaded SourceFile
      * @returns Collection of candidate Lambda handler information, empty array otherwise
      */
-    private static visitSourceFileChild(node: ts.Node, filename: string): LambdaHandlerCandidate[] {
+    private processSourceFile(sourceFile: ts.SourceFile, filename: string): LambdaHandlerCandidate[] {
+        // functionDeclarations - each top level function declared
+        const functionDeclarations: ts.SignatureDeclaration[] = []
+        // expressionStatements - all statements like "exports.handler = ..."
+        const expressionStatements: ts.ExpressionStatement[] = []
+        // exportNodes - all "export function Xyz()"
+        const exportNodes: ts.Node[] = []
+
+        // look for nodes of interest
+        sourceFile.forEachChild((node: ts.Node) => {
+            if (ts.isFunctionLike(node)) {
+                functionDeclarations.push(node)
+            }
+
+            if (ts.isExpressionStatement(node) && (ts.isBinaryExpression(node.expression))) {
+                expressionStatements.push(node)
+            }
+
+            if (TypescriptLambdaHandlerSearch.isNodeExported(node)) {
+                exportNodes.push(node)
+            }
+        })
+
         const handlers: LambdaHandlerCandidate[] = []
 
-        if (this.isNodeExported(node)) {
-            const baseFilename = path.parse(filename).name
+        handlers.push(...TypescriptLambdaHandlerSearch.findJavascriptHandlerCandidates(
```

Comment:
>     // exportNodes - all "export function Xyz()"

The difference between `findJavascriptHandlerCandidates` and `findTypescriptHandlerCandidates` has nothing to do with JavaScript vs TypeScript. The `export function Xyz()` syntax was added in ES6 (and was supported by Webpack prior to ES6), and is valid in both JS and TS.

Reply:
Ok, I'll name them more plainly